stdlib/task/src/main/java/org/ballerinalang/stdlib/task/utils/Utils.java

```diff
@@ -0,0 +1,135 @@
+/*
+ *  Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ *
+ */
+package org.ballerinalang.stdlib.task.utils;
+
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.bre.bvm.BLangVMErrors;
+import org.ballerinalang.connector.api.BLangConnectorSPIUtil;
+import org.ballerinalang.connector.api.Resource;
+import org.ballerinalang.connector.api.Service;
+import org.ballerinalang.model.types.BTypes;
+import org.ballerinalang.model.values.BError;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.stdlib.task.exceptions.SchedulingException;
+import org.ballerinalang.util.codegen.FunctionInfo;
+import org.ballerinalang.util.exceptions.BLangRuntimeException;
+
+import java.util.Objects;
+
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.FIELD_DAYS_OF_MONTH;
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.FIELD_DAYS_OF_WEEK;
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.FIELD_HOURS;
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.FIELD_MINUTES;
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.FIELD_MONTHS;
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.FIELD_SECONDS;
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.FIELD_YEAR;
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.PACKAGE_STRUCK_NAME;
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.RESOURCE_ON_TRIGGER;
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.STRUCT_APPOINTMENT_DATA;
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.TASK_ERROR_CODE;
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.TASK_ERROR_MESSAGE;
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.TASK_ERROR_RECORD;
+import static org.quartz.CronExpression.isValidExpression;
+
+/**
+ * Utility functions used in ballerina task module.
```

Comment:
Please add `@since` annotation for all the new classes

Reply:
Done.